### PR TITLE
HIVE-26044: Remove hardcoded version references from the tests

### PR DESCRIPTION
**Pull request by pvary**
### What changes were proposed in this pull request?
Remove hardcoded version references from the tests

### Why are the changes needed?
For easier version changes

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Running the unit tests

### CODE REVIEW

File: itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java [hunk 1]
```diff
@@ -287,7 +291,7 @@ public void test() throws Exception {
 
     List<String> args = new ArrayList<String>(baseArgs);
     args.add("-f");
-    args.add("../../metastore/scripts/upgrade/hive/hive-schema-4.0.0-alpha-1.hive.sql");
+    args.add("../../metastore/scripts/upgrade/hive/hive-schema-" + hiveSchemaVer + ".hive.sql");
```
Comment (marton-bod) [inline itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java:294]: Can we use `HiveVersionInfo.getShortVersion()` in these places too?

**Review by marton-bod — APPROVED** [review]
LGTM, just one question

**Review by zabetak — COMMENTED** [review]
LGTM. Although I have the same comment with Marton. Can't we use `HiveVersionInfo` everywhere?

File: itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java (hunk 1 above)
Comment (pvary) [reply itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java:294]: I think that the  `HiveVersionInfo.getShortVersion()` and the `MetaStoreSchemaInfoFactory.get(miniHS2.getServerConf()).getHiveSchemaVersion()` might and should be different in several deployments. Some random vendor (😄) might release different code version (hotfixes, maintenance releases) for the same HMS DB version.

I think this is why `MetaStoreSchemaInfoFactory` has been created, and I tried to use the appropriate version in the appropriate place.

Comment (marton-bod) [reply itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java:294]: Oh I see. Then I think you did it correctly

Comment (zabetak) [reply itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java:294]: So far we have:
- `HiveVersionInfo`
- `MetastoreVersionInfo`
- `MetaStoreSchemaInfo`
I understand there is a concern between HS2 and HMS version being different but what about `MetastoreVersionInfo`
vs `MetaStoreSchemaInfo`?

Comment (pvary) [reply itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java:294]: * `HiveVersionInfo` is the code version of the HS2
* `MetastoreVersionInfo` is the code version of the HMS
* `MetaStoreSchemaInfo` is the db schema version under the HMS - some HMS changes do not require schema changes

Comment (zabetak) [reply itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java:294]: Thanks both for the clarifications!

**Review by zabetak — APPROVED** [review]

**Pull request merged**